src/providers/export/tensorFlowPascalVOC.ts

```diff
@@ -155,16 +197,147 @@ item {
         // Create ImageSets Sub Folder (Main ?)
         const imageSetsFolderName = `${exportFolderName}/ImageSets`;
         await this.storageProvider.createContainer(imageSetsFolderName);
 
-        // Save ImageSets (Main ?)
-        // TODO
+        const imageSetsMainFolderName = `${exportFolderName}/ImageSets/Main`;
+        await this.storageProvider.createContainer(imageSetsMainFolderName);
+
+        const tagsDict = new Map<string, string[]>();
+        if (tags) {
+            tags.forEach((tag) => {
+                tagsDict.set(tag.name, []);
+            });
+
+            allAssets.forEach((asset) => {
+                if (asset.regions.length > 0) {
+                    asset.regions.forEach((region) => {
+                        tags.forEach((tag) => {
+                            const array = tagsDict.get(tag.name);
+                            if (region.tags.filter((regionTag) => regionTag.name === tag.name).length > 0) {
+                                array.push(`${asset.asset.name} 1`);
+                            } else {
+                                array.push(`${asset.asset.name} -1`);
+                            }
+                        });
+                    });
+                } else if (exportUnassignedTags) {
+                    tags.forEach((tag) => {
+                        const array = tagsDict.get(tag.name);
+                        array.push(`${asset.asset.name} -1`);
+                    });
+                }
+            });
+
+            // Save ImageSets
+            tags.forEach(async (tag) => {
+                if (testSplit > 0 && testSplit <= 1) {
+                    // Shuffle tagsDict sets
+                    tagsDict.forEach((value, key) => {
+                        value = this.shuffle(value);
+                    });
+
+                    const array = tagsDict.get(tag.name);
+
+                    // Split in Test and Train sets
+                    const totalAssets = array.length;
+                    const testCount = Math.ceil(totalAssets * testSplit);
+
+                    const testArray = array.slice(0, testCount);
+                    const trainArray = array.slice(testCount, totalAssets);
+
+                    const testImageSetFileName = `${imageSetsMainFolderName}/${tag.name}_val.txt`;
+                    await this.storageProvider.writeText(testImageSetFileName, testArray.join("\n"));
+
+                    const trainImageSetFileName = `${imageSetsMainFolderName}/${tag.name}_train.txt`;
+                    await this.storageProvider.writeText(trainImageSetFileName, trainArray.join("\n"));
+
+                } else {
+                    const imageSetFileName = `${imageSetsMainFolderName}/${tag.name}.txt`;
+                    await this.storageProvider.writeText(imageSetFileName, tagsDict.get(tag.name).join("\n"));
+                }
+            });
+        }
+    }
+
+    private shuffle(a: any[]) {
```

Comment:
What does this do and how is it used?

Reply:
shuffle() it's a utility function to shuffle a generic array.  I call this method to shuffle each tag arrays before splitting in test and validation set.  In ML if you're not working with sequenced datapoint like for example time series it's a good practice to shuffle an input dataset before splitting it in test and training dataset.

Anyway I'll put some comment in the code